### PR TITLE
use submission api v5 to download submission

### PR DESCRIPTION
**Pull request by deedee**
new token property
`@submission_api_endpoint@=http://localhost:3000/api/v5`

### CODE REVIEW

**Comment by callmekatootie** [comment]
@deedee Was this closed intentionally? Is there an alternative PR?

**Comment by deedee** [comment]
@callmekatootie I thought this task was cancel. Do you want me to reopen this?

**Comment by callmekatootie** [comment]
@deedee Checking... they asked me to review this but I see now that its closed :smile: 

Will let you know

**Comment by callmekatootie** [comment]
Hey @deedee - there's been a misunderstanding. This task is very much active. Could you kindly reopen this and let me know if it is ready to be reviewed?

**Comment by deedee** [comment]
@callmekatootie Yes it's ready for review

File: src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java [hunk 1]
```diff
@@ -36,11 +15,7 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.topcoder.management.deliverable.Submission;
-import com.topcoder.management.deliverable.SubmissionStatus;
-import com.topcoder.management.deliverable.Upload;
-import com.topcoder.management.deliverable.UploadManager;
-import com.topcoder.management.deliverable.UploadStatus;
+import com.topcoder.management.deliverable.*;
```
Comment (skyhit) [inline src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java]: @deedee let's avoid to useing import *.

Comment (deedee) [reply src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java]: @skyhit Ok. intellij done this :)

File: src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java [hunk 2]
```diff
@@ -57,23 +32,28 @@
...
 import com.topcoder.search.builder.filter.EqualToFilter;
 import com.topcoder.search.builder.filter.Filter;
-import com.topcoder.servlet.request.ConfigurationException;
-import com.topcoder.servlet.request.DisallowedDirectoryException;
-import com.topcoder.servlet.request.FileDoesNotExistException;
-import com.topcoder.servlet.request.FileUpload;
-import com.topcoder.servlet.request.FileUploadResult;
-import com.topcoder.servlet.request.PersistenceException;
-import com.topcoder.servlet.request.UploadedFile;
+import com.topcoder.servlet.request.*;
```
Comment (skyhit) [inline src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java]: @deedee let's avoid to useing import *.

File: src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java [hunk 3]
```diff
@@ -57,23 +32,28 @@
...
 import com.topcoder.util.file.Template;
-import com.topcoder.util.file.fieldconfig.Condition;
 import com.topcoder.util.file.fieldconfig.Field;
 import com.topcoder.util.file.fieldconfig.Node;
 import com.topcoder.util.file.fieldconfig.TemplateFields;
 import com.topcoder.util.file.templatesource.FileTemplateSource;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
```
Comment (skyhit) [inline src/java/main/com/cronos/onlinereview/actions/projectdetails/BaseProjectDetailsAction.java]: @deedee let's avoid to useing import *.

**Comment by callmekatootie** [comment]
@deedee Could you kindly make the changes requested by @skyhit and let me know once that is ready for review?

**Comment by deedee** [comment]
@callmekatootie change done on new commit.

**Comment by callmekatootie** [comment]
@skyhit Are the updates fine?

**Comment by skyhit** [comment]
@deedee sorry, foget to submit some new comments, please check again

**Comment by callmekatootie** [comment]
@deedee Sorry about that - could you kindly check out @skyhit 's review one more time and carry out the changes as needed.

Also - can you let ping me on slack on the same handle as my github...

File: src/java/main/com/cronos/onlinereview/util/ActionsHelper.java [hunk 4]
```diff
@@ -131,6 +86,33 @@
...
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.net.URI;
+import java.net.URL;
+import java.rmi.RemoteException;
+import java.sql.*;
+import java.text.Format;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (deedee) [reply src/java/main/com/cronos/onlinereview/util/ActionsHelper.java]: second commit already fixed this

File: src/java/main/com/cronos/onlinereview/util/ActionsHelper.java [hunk 5]
```diff
@@ -131,6 +86,33 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.net.URI;
+import java.net.URL;
+import java.rmi.RemoteException;
+import java.sql.*;
```
[the remark this answers is not in the document]
Comment (deedee) [reply src/java/main/com/cronos/onlinereview/util/ActionsHelper.java]: second commit already fixed this

File: src/java/main/com/cronos/onlinereview/util/ActionsHelper.java [hunk 6]
```diff
@@ -113,11 +72,7 @@
 import com.topcoder.project.phases.template.StartDateGenerator;
 import com.topcoder.project.phases.template.persistence.XmlPhaseTemplatePersistence;
 import com.topcoder.search.builder.SearchBuilderException;
-import com.topcoder.search.builder.filter.AndFilter;
-import com.topcoder.search.builder.filter.EqualToFilter;
-import com.topcoder.search.builder.filter.Filter;
-import com.topcoder.search.builder.filter.NotFilter;
-import com.topcoder.search.builder.filter.OrFilter;
+import com.topcoder.search.builder.filter.*;
```
[the remark this answers is not in the document]
Comment (deedee) [reply src/java/main/com/cronos/onlinereview/util/ActionsHelper.java]: second commit already fixed this

File: src/java/main/com/cronos/onlinereview/util/ActionsHelper.java [hunk 7]
```diff
@@ -58,21 +23,15 @@
...
-import com.topcoder.db.connectionfactory.DBConnectionException;
-import com.topcoder.db.connectionfactory.DBConnectionFactory;
-import com.topcoder.db.connectionfactory.DBConnectionFactoryImpl;
-import com.topcoder.db.connectionfactory.UnknownConnectionException;
-import com.topcoder.management.deliverable.Deliverable;
-import com.topcoder.management.deliverable.DeliverableManager;
-import com.topcoder.management.deliverable.Submission;
-import com.topcoder.management.deliverable.Upload;
-import com.topcoder.management.deliverable.UploadManager;
+import com.topcoder.db.connectionfactory.*;
```
[the remark this answers is not in the document]
Comment (deedee) [reply src/java/main/com/cronos/onlinereview/util/ActionsHelper.java]: second commit already fixed this

**Comment by deedee** [comment]
@callmekatootie @skyhit new change has been pushed

**Comment by skyhit** [comment]
@deedee @callmekatootie from code review, looks good to me.

**Comment by callmekatootie** [comment]
Ticket https://github.com/topcoder-platform/tc-online-review/issues/89

**Comment by callmekatootie** [comment]
Current Status:

- Code review completed.
- However, due to steep deployment process, could not deploy and verify.
- Suggestion is to merge the PR and test directly on topcoder's dev instance. Awaiting approval.

**Pull request closed without merging**